benchexec/tools/coveriteam.py

```diff
@@ -0,0 +1,72 @@
+# This file is part of BenchExec, a framework for reliable benchmarking:
+# https://github.com/sosy-lab/benchexec
+#
+# SPDX-FileCopyrightText: 2007-2020 Dirk Beyer <https://www.sosy-lab.org>
+#
+# SPDX-License-Identifier: Apache-2.0
+
+import benchexec.tools.template
+import benchexec.util as util
+import benchexec.result as result
+
+
+class Tool(benchexec.tools.template.BaseTool):
+    """
+    Tool info for CoVeriTeam: On-Demand Composition of Cooperative Verification Systems.
+    URL: https://gitlab.com/sosy-lab/software/coveriteam.
+
+    This class has 2 purposes:
+        1. to serve as an abstract class for specific coveriteam programs like verifiers, validators, etc.
+        2. to serve as the tool info module for any generic coveritea program.
```

Comment:
If this is not only intended to be an abstract class, it needs `name()`.

But later on there are comments that further contradict this. So what is actually the intention?

Wouldn't it make much more sense to have just one module for CoVeriTeam? Otherwise there will probably be lots of modules, BenchExec needs to be changed every time CoVeriTeam gets a new use case, and the names of these are not really nice as can be seen already.